### PR TITLE
Added HowTo guide for creating a plugin with the MinecraftDev IJ plugin

### PR DESCRIPTION
**Pull request by Leguan16**
As communicated in the discord I took the initiative to add a tutorial for creating a project with help of the Minecraft Development IntelliJ plugin. 

I did not yet take a look at typos or grammar and just want to gather some feedback for the overall structure. 
I also changed the current `project-setup` markdown file to `manual-project-setup`. 
The `new_project_filled.png` currently is not in use. This might or might not be added later. 
For the list to explain the fields when creating a new project I am not sure if we should use a table instead. 

Feedback is appreciated!

### CODE REVIEW

File: docs/paper/dev/getting-started/manual-project-setup.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-slug: /dev/project-setup
+slug: /dev/manual-project-setup
```
Comment (powercasgamer) [inline docs/paper/dev/getting-started/manual-project-setup.md:2]: redirect might be required here?

File: docs/paper/dev/getting-started/assisted-project-setup.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* Location: The location of your project. This is where the project files will be stored.
+* Platform Type: The platform type you are developing for. This should be `Plugin`.
+* Platform: The platform you are developing for. This should be `Paper`.
+* Minecraft Version: The version of Minecraft you are developing for.
+* Plugin Name: The name of your plugin.
+* Main Class: The main class of your plugin. This should be the class that extends `JavaPlugin`.
+* Optional Settings: Here you can define things like authors, website, description, etc. These are optional and not required for the plugin to work.
+* Build System: The build system you want to use. Paper recommends using Gradle but you can use Maven if you prefer.
+* Paper manifest: Whether you want to use the new Paper Plugins or not. For now this is not recommended as it is still in development.
+* Group ID: The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `com.<yourname>`
```
Comment (powercasgamer) [inline docs/paper/dev/getting-started/assisted-project-setup.md:42]: You should use a domain you own, so something like `io.github.<username>.<project>` or `me.<username>` is also commonly used

File: docs/paper/dev/getting-started/assisted-project-setup.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+![](./assets/plugin_overview.png)
+
+### Conclusion
+
+You should now have a project set up with Paper as a dependency.
+All you have left to do now is to compile your plugin and run it on a Paper server.
+
+:::note
+
+If you want to streamline the process of testing a plugin, you can use the [Run-Paper](https://github.com/jpenilla/run-paper) Gradle task.
```
Comment (powercasgamer) [inline docs/paper/dev/getting-started/assisted-project-setup.md:58]:
```suggestion
If you want to streamline the process of testing a plugin, you can use the [Run-Paper]https://github.com/jpenilla/run-task) Gradle plugin.
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]

File: docs/paper/dev/getting-started/assisted-project-setup.md (hunk 2 above)
Comment (Leguan16) [reply docs/paper/dev/getting-started/assisted-project-setup.md:42]: but you most likely don't own me.<name> and not everybody has a GitHub account. I don't think this is to change.

File: docs/paper/dev/getting-started/manual-project-setup.md (hunk 1 above)
Comment (Leguan16) [reply docs/paper/dev/getting-started/manual-project-setup.md:2]: from to? i cannot follow.

File: docs/paper/dev/getting-started/assisted-project-setup.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
+---
+slug: /dev/assisted-project-setup
+description: Step-by-step instructions on how to setup a plugin with the MinecraftDev plugin.
```
Comment (olijeffers0n) [inline docs/paper/dev/getting-started/assisted-project-setup.md:3]: Don’t like this slug

File: docs/paper/dev/getting-started/assisted-project-setup.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+# Paper Project setup with the MinecraftDev IntelliJ plugin
+If you are new to plugin development you might be a little overwhelmed by the amount of things you need to do to get started.
+This guide aims to help you get started with setting up a new project using the MinecraftDev IntelliJ plugin.
+
+:::note
+
+This tutorial only works with IntelliJ IDEA. If you are using another IDE, please follow the [manual project setup guide](./manual-project-setup).
+
+:::
```
Comment (olijeffers0n) [inline docs/paper/dev/getting-started/assisted-project-setup.md:14]: new line please

File: docs/paper/dev/getting-started/manual-project-setup.md (hunk 1 above)
Comment (olijeffers0n) [reply docs/paper/dev/getting-started/manual-project-setup.md:2]: yes, you cannot just change slugs. You need to set a redirect in the vercel file

File: docs/paper/dev/getting-started/manual-project-setup.md [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
 description: Step-by-step instructions on how to setup a plugin development environment.
 ---
 
-# Paper Project Setup
+# Manual Paper Project Setup
```
Comment (olijeffers0n) [inline docs/paper/dev/getting-started/manual-project-setup.md:6]: Bleh, don’t like manual. Just leave it as project setup and then have the mcdev example

**Review by olijeffers0n — CHANGES_REQUESTED** [review]

File: docs/paper/dev/getting-started/manual-project-setup.md (hunk 6 above)
Comment (Leguan16) [reply docs/paper/dev/getting-started/manual-project-setup.md:6]: alright

**Comment by Leguan16** [comment]
Alright,
I changed a few things:

- I moved `project-setup` and `mcdev-project-setup` to a separate subsection `project-setup`
- I changed the list to a table
- I reverted the slug

Now, with those changes I don't think we should leave the current project setup tutorial named "Paper Project Setup" since it should be obvious that we try to create a paper project. And it would look better if its just named "Project Setup" and the other one is called "McDev Project Setup". Maybe even just rename the title to "Manual Project Setup" and leave the slug how it is?

**Comment by olijeffers0n** [comment]
Yeah so this is ready for content reviews - i’m happy with the addition of a new folder but you need to add a redirect for the slug change :)

**Comment by Leguan16** [comment]
> Yeah so this is ready for content reviews - i’m happy with the addition of a new folder but you need to add a redirect for the slug change :)

Dang it i forgot it again xd

**Comment by Leguan16** [comment]
I think i changed it?

File: docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
+---
+slug: /dev/getting-started/project-setup/mcdev-project-setup
+description: Step-by-step instructions on how to setup a plugin with the MinecraftDev plugin.
```
Comment (olijeffers0n) [inline docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md:3]: I think clarifying the two “plugin”s is needed

**Review by olijeffers0n — CHANGES_REQUESTED** [review]
Just one thing really, redirect lgtm

File: docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md (hunk 7 above)
Comment (Leguan16) [reply docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md:3]: Something like this?
```suggestion
description: Step-by-step instructions on how to setup a plugin with the MinecraftDev IntelliJ plugin.
```

Comment (olijeffers0n) [reply docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md:3]: maybe change the first plugin to project and keep that Intelljj

Comment (Leguan16) [reply docs/paper/dev/getting-started/project-setup/mcdev-project-setup.md:3]: Ok I changed it. Will push when I have more feedback.

**Comment by Leguan16** [comment]
I moved the mcdev stuff over to misc and added a section for velocity plugins.
I also reverted the manual project setup back to its previous state.

**Comment by olijeffers0n** [comment]
My apologies, that wasn't quite what I meant by misc. I meant `Paper/Dev/Miscellaneous` - Sorry

File: docs/misc/mcdev-project-setup.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+| **Optional Settings** | Here you can define things like authors, website, description, etc. These are optional and not required for the plugin to work.                                                                                                                                           |
+| **Build System**      | The build system you want to use. Paper recommends using Gradle but you can use Maven if you prefer.                                                                                                                                                                      |
+| **Paper Manifest**    | Whether you want to use the new Paper plugins or not. For now this is not recommended as it is still in development.                                                                                                                                                      |
+| **Group ID**          | The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `io.github.<yourname>` or if you don't have GitHub you can just use `me.<yourname>`. |
```
Comment (powercasgamer) [inline docs/misc/mcdev-project-setup.md:48]:
```suggestion
| **Group ID**          | The group ID of your project. This is used for Maven and Gradle. This is usually your domain name in reverse. If you don't know what you should put here, you can use something like `io.github.<yourname>` or if you don't have GitHub you can use `me.<yourname>`. |
```

**Review by powercasgamer — CHANGES_REQUESTED** [review]

**Review by olijeffers0n — APPROVED** [review]

**Review by olijeffers0n — APPROVED** [review]

**Comment by olijeffers0n** [comment]
Just needs a merge from main please 🙏

**Pull request merged**